src/fields/fp.rs

```diff
@@ -669,68 +669,78 @@ lazy_static! {
     static ref FP_TABLES: SqrtTables<Fp> = SqrtTables::new(0x11BE, 1098);
 }
 
+#[cfg(feature = "std")]
```

Comment:
Why does this need std?

Reply:
It needs `alloc`, but in #4 I made everything require `std` because `FieldExt` used `std::io`. However, after this PR `std::io` isn't required, so we _might_ be able to lower the requirement to `alloc`.

Reply:
Nope, the `CurveAffine` extension trait also relies on `std::io`. I'll figure out whether we can delete that as well.

Reply:
We probably can, but let's do this in a subsequent PR.